### ISSUE TITLE
Google-Vertex Tool Call Not Recognized in @ai-sdk/google-vertex v0.0.27+

### ISSUE THREAD

**Issue by harithoppil** [vercel/ai#2715]
### Description

I'm encountering an issue with the Google Vertex Provider tool calling functionality when attempting an update of `@ai-sdk/google-vertex`.  

1. I am able to reproduce the issue example code from `examples/ai-core/src/generate-text`  file `google-vertex-tool-call.ts`, 
2. I have narrowed down the issue starting from version 0.0.27 and persists in later versions up to 0.0.30.  
3. I have attempted to find a workaround without success  with and without roundtrips, Explicitly setting useSearchGrounding , using prompt vs messages as input, changing `@google-cloud/vertexai` versions. 

### Observed Behavior
- **Version 0.0.26 (Working):**
  Output: `Getting weather for New York City The weather in New York City is 76 degrees.`
    
- **Version 0.0.27+ (Not Working):**
  Output: `I do not have access to real-time information, including weather conditions. To get the current weather in New York City, I recommend checking a reliable weather app or website.`
    
### Reproduction Steps
1. Clone the Vercel AI SDK repository and set up the example:
```
git clone https://github.com/vercel/ai.git
cd ai
pnpm install
pnpm build
cd examples/ai-core
```
    
4. Create a `.env` file and configure the following variables:
- `GOOGLE_VERTEX_LOCATION`
- `GOOGLE_VERTEX_PROJECT`

5.  Run the example code with `@ai-sdk/google-vertex` version 0.0.30:
    
    `pnpm tsx src/generate-text/google-vertex-tool-call.ts`

### Code example
Unmodified google-vertex tool call using generateText Example from ai-core 
[https://github.com/vercel/ai/blob/main/examples/ai-core/src/generate-text/google-vertex-tool-call.ts](url)

```
import { vertex } from '@ai-sdk/google-vertex';
import { generateText, tool } from 'ai';
import dotenv from 'dotenv';
import { z } from 'zod';

dotenv.config();

async function main() {
  const { text } = await generateText({
    model: vertex('gemini-1.5-pro'),
    prompt: 'What is the weather in New York City? ',
    tools: {
      weather: tool({
        description: 'Get the weather in a location',
        parameters: z.object({
          location: z.string().describe('The location to get the weather for'),
        }),
        execute: async ({ location }) => {
          console.log('Getting weather for', location);
          return {
            location,
            temperature: 72 + Math.floor(Math.random() * 21) - 10,
          };
        },
      }),
    },
    maxToolRoundtrips: 5,
  });

  console.log(text);
}

main().catch(console.error);
```
### Additional Notes
- The issue persists in versions 0.0.27 through 0.0.30 of `@ai-sdk/google-vertex`.
- No error messages are displayed in the console.

**Comment by lgrammel** [comment]
Fix: https://github.com/vercel/ai/pull/2716/files

**Comment by lgrammel** [comment]
Thanks for the bug report. Fixed in `@ai-sdk/google-vertex@0.0.31`

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This is a concrete regression in existing Google Vertex tool-calling behavior, confirmed and fixed by a maintainer.

### Evidence
- Tool calling worked in @ai-sdk/google-vertex 0.0.26 but regressed in versions 0.0.27–0.0.30.
- The provided repository example failed to invoke the weather tool and instead returned a generic no-real-time-information response.
- A maintainer explicitly called it a bug report and confirmed a fix in @ai-sdk/google-vertex@0.0.31.
